src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -6018,16 +6020,20 @@ struct MulSimplify
   LogicalResult matchAndRewriteImpl(stablehlo::MulOp op,
                                     PatternRewriter &rewriter) const {
 
-    // 0 * x -> x
+    // 0 * x -> 0
     if (matchPattern(op.getLhs(), m_AnyZeroFloat()) ||
-        matchPattern(op.getLhs(), m_Zero())) {
-      rewriter.replaceOp(op, op.getLhs());
+            matchPattern(op.getLhs(), m_Zero()),
+        matchPattern(op.getLhs(), m_AnyZeroComplex())) {
+      rewriter.replaceOpWithNewOp<stablehlo::ConstantOp>(
+          op, cast<ElementsAttr>(makeAttr(op.getType(), 0)));
       return success();
     }
-    // x * 0 -> x
+    // x * 0 -> 0
     if (matchPattern(op.getRhs(), m_AnyZeroFloat()) ||
-        matchPattern(op.getRhs(), m_Zero())) {
-      rewriter.replaceOp(op, op.getRhs());
+            matchPattern(op.getRhs(), m_Zero()),
```

Comment:
we should also add a similar anyzerocomplex to the enzyme-simplify-math infra as well

Reply:
see https://github.com/EnzymeAD/Enzyme/blob/0828c872bd7ceb5cd3ae0a7e8bbea28f26d8ecad/enzyme/Enzyme/MLIR/Implementations/ArithAutoDiffOpInterfaceImpl.cpp#L64 for example (and relevant here)

Reply:
https://github.com/EnzymeAD/Enzyme/blob/0828c872bd7ceb5cd3ae0a7e8bbea28f26d8ecad/enzyme/Enzyme/MLIR/Implementations/ComplexAutoDiffOpInterfaceImpl.cpp#L50 seems to implement these